CONTRIBUTING.md

```diff
@@ -0,0 +1,172 @@
+    with the previous behavior
+
+## Submission
+
+### Pre-requisites
+
+- Make sure you have a [GitHub account](https://github.com/join)
+
+- [Open an
+  issue](https://github.com/puppetlabs/puppetlabs-http_request/issues/new/choose)
+  or [track an
+  issue](https://github.com/puppetlabs/puppetlabs-http_request/issues) you are
+  patching for
+
+### Push and PR
+
+- Push your changes to your fork
+
+- [Open a Pull
+  Request](https://github.com/puppetlabs/puppetlabs-http_request/compare)
+  against the repository
+
+## More about commits 
+
+### Make separate commits for logically separate changes.
+
+Please break your commits down into logically consistent units which include new
+or changed tests relevant to the rest of the change.  The goal of doing this is
+to make the diff easier to read for whoever is reviewing your code.  In general,
+the easier your diff is to read, the more likely someone will be happy to review
+it and get it into the code base.
+
+If you are going to refactor a piece of code, please do so as a separate commit
+from your feature or bug fix changes.
+
+We also really appreciate changes that include tests to make sure the bug is not
+re-introduced, and that the feature is not accidentally broken.
+
+Describe the technical detail of the change(s).  If your description starts to
+get too long, that is a good sign that you probably need to split up your commit
+into more finely grained pieces.
+
+Commits which plainly describe the things which help reviewers check the patch
+and future developers understand the code are much more likely to be merged in
+with a minimum of bike-shedding or requested changes.  Ideally, the commit
+message would include information, and be in a form suitable for inclusion in
+the release notes for the version of Puppet that includes them.
+
+Please also check that you are not introducing any trailing whitespace or other
+"whitespace errors".  You can do this by running `git diff --check` on your
+changes before you commit.
+
+### Sending your patches
+
+To submit your changes via a GitHub pull request, we _highly_ recommend that you
+have them on a topic branch, instead of directly on "main". It makes things much
+easier to keep track of, especially if you decide to work on another thing
+before your first change is merged in.
+
+GitHub has some pretty good [general documentation](http://help.github.com/) on
+using their site.  They also have documentation on [creating pull
+requests](https://help.github.com/articles/creating-a-pull-request-from-a-fork/).
+
+In general, after pushing your topic branch up to your repository on GitHub, you
+can switch to the branch in the GitHub UI and click "Pull Request" towards the
+top of the page in order to open a pull request.
+
+### Update the related GitHub issue.
+
+If there is a GitHub issue associated with the change you submitted, link the
+issue to your pull request.
+
+## Testing
```

Comment:
I'd also move this to be between 'Commit Checklist' and 'Submission'. This is honestly probably what most people are looking for - they know how to use Git, but just need the commands to run the tests.